java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/AutoStopEmrServerlessApplicationLambdaIT.java

```diff
@@ -0,0 +1,169 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.cdk.custom;
+
+import com.amazonaws.services.lambda.runtime.events.CloudFormationCustomResourceEvent;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.emrserverless.model.JobRunState;
+
+import sleeper.core.util.PollWithRetries;
+
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.aResponseWithJobRunWithState;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.aResponseWithNoJobRuns;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.aResponseWithStoppingApplication;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.aResponseWithTerminatedApplication;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.anyRequestedForEmrServerless;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.cancelJobRunRequest;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.cancelJobRunRequested;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.getApplicationRequest;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.getApplicationRequested;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.listRunningJobsForApplicationRequest;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.listRunningJobsForApplicationRequested;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.listRunningOrCancellingJobsForApplicationRequest;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.listRunningOrCancellingJobsForApplicationRequested;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.stopApplicationRequest;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.stopApplicationRequested;
+import static sleeper.cdk.custom.WiremockEmrServerlessTestHelper.wiremockEmrServerlessClient;
+
+@WireMockTest
+public class AutoStopEmrServerlessApplicationLambdaIT {
+
+    private AutoStopEmrServerlessApplicationLambda lambda;
+    private String applicationId;
+
+    @BeforeEach
+    void setUp() {
+        applicationId = "test-app-id";
```

Comment:
The application ID is hard coded in the JSON set in the helpers. It might be easier to see which request/responses have it if we pass it into the helper methods?